src/adjacently.rs

```diff
@@ -133,11 +136,26 @@ impl<'de, T: ?Sized> Visitor<'de> for TaggedVisitor<T> {
                         return Err(de::Error::duplicate_field(self.content));
                     }
                     // There is no second key.
-                    None => return Err(de::Error::missing_field(self.tag)),
+                    None => {
+                        if let Some(variant) = self.default_variant {
+                            // No variant is specified but there is a default variant.
+                            if let Some(Some(deserialize_fn)) = self.registry.map.get(variant) {
+                                let fn_apply = FnApply { deserialize_fn: *deserialize_fn };
+                                let content = content.into_deserializer();
+                                fn_apply.deserialize(content)?
+                            } else {
+                                // The default variant is not found in the registry.
+                                return Err(de::Error::unknown_variant(variant, &self.registry.names));
+                            }
+                        } else {
+                            // No variant is specified and there is no default variant.
+                            return Err(de::Error::missing_field(self.tag));
+                        }
+                    }
                 }
             }
             // There is no first key.
-            None => return Err(de::Error::missing_field(self.tag)),
+            None => return Err(de::Error::missing_field(self.content)),
```

Comment:
I would prefer if this would continue to return missing field `self.tag` on traits which do not have a default variant.